system-test/kitchen.ts

```diff
@@ -0,0 +1,184 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as assert from 'assert';
+import {describe, it, beforeEach} from 'mocha';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import {Readable} from 'stream';
+import {createURI, ErrorWithCode, upload} from '../src/gcs-resumable-upload';
+
+const bucketName = process.env.BUCKET_NAME || 'gcs-resumable-upload-test';
+const filePath = path.join(__dirname, '../../system-test/data/20MB.zip');
```

Comment:
If we can, let's remove this binary in favor of generating a temp file. See:
- https://nodejs.org/api/os.html#ostmpdir
- https://nodejs.org/api/fs.html#fspromisesmkdtempprefix-options

If we want a 20MB file of random bytes:
- `crypto.randomBytes(1024 * 1024 * 20)`

This way we don't have to check in a test blob into the repo.

Reply:
Good call. I would agree as well, I'm not a huge fan of committing blobs.